File: Java/Array/Fitting the array.java
import java.util.Arrays;

// Problem Link: https://practice.geeksforgeeks.org/problems/fitting-the-array1514/1


/***
 * Approach: we have to rearrange elements such that all array arr elements
 * fits in brr array.
 * For this, I sort both arrays in ascending order and check them.
 */

/***
 * Time Complexity: O(n log n)
 * Space Complexity: O(1)
 */



class Solution{
    
   
    // Function for finding maximum and value pair
    public static boolean isFit (int arr[], int brr[], int n) {
        //Complete the function
        Arrays.sort(arr);
        Arrays.sort(brr);
        for (int i=0;i<n;i++)
        {
            if(arr[i]>brr[i])
            {
                return false;
            }
        }
        return true;
    }
    
}

File: C++/Array/Fitting the Array.cpp
//Problem Link: https://practice.geeksforgeeks.org/problems/fitting-the-array1514/1

/*
Time Complexity: O(N*log(N))
Space Complexity: O(1)
*/

/*
Approach: 
Count = No. of elemnt which satisfy the given condition
n = Size od an array

Step 1: Sort both the array.
Step 2: Check b[i]>=a[i] (Assign the minimum element of array A[] to the minimun element of array b[]) if it is true then increment count by 1. If this condition fail at any indeex then return false.
Step 3: Check count==n if true return 1, else return 0 ;
*/

class Solution{
public:    
    bool isFit(int arr[], int brr[], int n){
        int count =0 ;
        sort(arr,arr+n) ;
        sort(brr,brr+n) ;
        for(int i=0;i<n;i++) {
            if(brr[i]>=arr[i])
                count++ ;
            else 
                return 0 ;
        }
        if(count==n){
            return 1;
        }
        return 0;
    }
};